configure succeeds but make fails when libzarith-ocaml but not libzarith-ocaml-dev is installed on ubuntu

<!-- Thank you for reporting a bug to Coq! -->

#### Description of the problem

#12939 is not sufficient, see https://github.com/HoTT/HoTT/pull/1392/checks?check_run_id=1048236906 where [`./configure` says](https://github.com/HoTT/HoTT/pull/1392/checks?check_run_id=1048236906#step:9:130) :
```
You have OCaml 4.05.0. Good!
You have OCamlfind 1.7.3. Good!
You have native-code compilation. Good!
You have the Zarith library installed. Good!
LablGtk3 or LablGtkSourceView3 not found:
=> no CoqIde will be built.

  Architecture                : Linux
  Sys.os_type                 : Unix
  Coq VM bytecode link flags  : -dllib -lcoqrun -dllpath /usr/local/lib/coq/kernel/byterun
  Other bytecode link flags   : 
  OCaml version               : 4.05.0
  OCaml binaries in           : /usr/bin/
  OCaml library in            : /usr/lib/ocaml
  OCaml flambda flags         : 
  Native dynamic link support : true
  CoqIde                      : no
  Documentation               : None
  Web browser                 : firefox -remote "OpenURL(%s,new-tab)" || firefox %s &
```
but [`make coqocaml` fails with](https://github.com/HoTT/HoTT/pull/1392/checks?check_run_id=1048236906#step:9:367)
```
OCAMLC    tools/coqdep.ml
OCAMLOPT  tools/coqdep.ml
OCAMLBEST -o bin/coqdep
File "_none_", line 1:
Error: Cannot find file /usr/lib/ocaml/zarith/zarith.cmxa
Makefile.build:501: recipe for target 'bin/coqdep' failed
```

cc @SkySkimmer 


#### Coq Version

94d9fe2b6cb06fe7f862683d8337433e18314001

--- a/configure.ml
+++ b/configure.ml
@@ -692,10 +692,13 @@ let operating_system =
 
 let check_for_zarith () =
   let zarith,_ = tryrun camlexec.find ["query";"zarith"] in
+  let zarith_cmai base = Sys.file_exists (base / "z.cmi") && Sys.file_exists (base / "zarith.cma") in
   let zarith_version, _ = run camlexec.find ["query"; "zarith"; "-format"; "%v"] in
   match zarith with
   | ""  ->
     die "Zarith library not installed, required"
+  | _ when not (zarith_cmai zarith) ->
+    die "Zarith library installed but no development files found (try installing the -dev package)"
   | _   ->
     let zarith_version_int = List.map int_of_string (numeric_prefix_list zarith_version) in
     if zarith_version_int >= [1;10;0] then

